src/Serializer/AbstractItemNormalizer.php

```diff
@@ -417,7 +489,7 @@ protected function getAttributeValue($object, $attribute, $format = null, array
      *
      * @return string|array
      */
-    private function normalizeRelation(PropertyMetadata $propertyMetadata, $relatedObject, string $resourceClass, string $format = null, array $context)
+    protected function normalizeRelation(PropertyMetadata $propertyMetadata, $relatedObject, string $resourceClass, string $format = null, array $context)
```

Comment:
I would prefer to don't propagate this data structure outside of the JSONAPI component. Can't you transform this array in IRI before passing it to the abstract normalizer?